sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/statsEstimation/FilterEstimationSuite.scala

```diff
@@ -821,6 +822,32 @@ class FilterEstimationSuite extends StatsEstimationTestBase {
       expectedRowCount = 3)
   }
 
+  test("ColumnStatsMap tests") {
```

Comment:
Since this is not correct, I think that FilterEstimation didn't work previously. Can you also add a related test using validateEstimatedStats similar to other tests? In the test, we should rely on min and max stats to do filter estimation.

Reply:
Hi @viirya , thanks for your review. Here the `FilterEstimation` works because we only call `FilterEstimation.estimate` to check it, in which we check if `rowCount.isEmpty` first and shortcut the estimation. If `rowCount` is not empty, we should always has a `min` and `max` in the stats, unless it is malformed.

Reply:
If `hasMinMaxStats` is not correct previously, will `evaluateBinary` work correctly?

Reply:
There's no public call to `evaluateBinary` method actually.

Reply:
Oh I see. So `FilterEstimation` works well without this change.